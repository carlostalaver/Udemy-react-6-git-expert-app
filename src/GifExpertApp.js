import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Goku']);
/* 
    const handleAdd = () => {
        setCategories(['HunterXHunter', ...categories]); // modificando el estado a traves de la propia referencia al array  categories
        setCategories(categ => ['HunterXHunter', ...categ]); // modificando el estado a treves un callback, para modificar el estado puedo usar cualquiera de las dos formas
    }; */

    return (
        <>
            <h2> GifExpertApp </h2>
            {/*  notar que le estoy pasando la referencia de setCategories al componente HIJO */}
            <AddCategory setCategories = { setCategories }/> 

           {/*  Esta seria una nueva forma de llamar al componente AddCategory */}
            {/* <AddCategory
                setCategories={(categories) => {
                    setCategories(categories);
                }}
            /> */}
            <hr/>
            <ol>
                {
                    categories.map((category) => ( // no se puede usar un for porque no es una expresion de js que retorne un valor
                        <GifGrid   // notar que no estoy usando un return xq uso un ( ) para sustituirlo
                            key={ category }  // se le asigna un key porque estoy iterando el componenete GifGrid
                            category= { category }
                        />
                    ))
                }
            </ol>
        </>
    )
}
