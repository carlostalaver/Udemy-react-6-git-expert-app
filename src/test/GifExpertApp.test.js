import React from 'react'
import { shallow } from 'enzyme'; 
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas del componente <GifExpertApp />', () => {

    test('Debe mostrarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />)

        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe mostrarse una lista de categorias', () => {

        const defaultCategories = ['One Punch', 'Goku'];
        
        const wrapper = shallow(<GifExpertApp defaultCategories = {defaultCategories}/>)

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( defaultCategories.length );
        
    })
    
    
})