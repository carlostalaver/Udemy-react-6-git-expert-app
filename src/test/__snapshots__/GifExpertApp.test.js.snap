// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> Debe mostrarse una lista de categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
